bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -0,0 +1,387 @@
+            .add("gx_charset", "UTF-8").add("SunQueryParamsString",
+                    Base64.getEncoder().encodeToString("realm=particuliers".getBytes(StandardCharsets.UTF_8)));
+
+    private static final int REFRESH_FIRST_HOUR_OF_DAY = 5;
+    private static final int REFRESH_INTERVAL_IN_MIN = 360;
+
+    private final OkHttpClient client = new OkHttpClient.Builder().followRedirects(false)
+            .cookieJar(new LinkyCookieJar()).build();
+    private final Gson gson = new Gson();
+
+    private @NonNullByDefault({}) ScheduledFuture<?> refreshJob;
+    private final WeekFields weekFields;
+
+    private final ExpiringDayCache<LinkyConsumptionData> cachedDaylyData;
+    private final ExpiringDayCache<LinkyConsumptionData> cachedMonthlyData;
+    private final ExpiringDayCache<LinkyConsumptionData> cachedYearlyData;
+
+    public LinkyHandler(Thing thing, LocaleProvider localeProvider) {
+        super(thing);
+        this.weekFields = WeekFields.of(localeProvider.getLocale());
+        this.cachedDaylyData = new ExpiringDayCache<LinkyConsumptionData>("daily cache", REFRESH_FIRST_HOUR_OF_DAY,
+                () -> {
+                    final LocalDate today = LocalDate.now();
+                    return getConsumptionData(DAILY, today.minusDays(13), today, true);
+                });
+        this.cachedMonthlyData = new ExpiringDayCache<LinkyConsumptionData>("monthly cache", REFRESH_FIRST_HOUR_OF_DAY,
+                () -> {
+                    final LocalDate today = LocalDate.now();
+                    return getConsumptionData(MONTHLY, today.withDayOfMonth(1).minusMonths(1), today, true);
+                });
+        this.cachedYearlyData = new ExpiringDayCache<LinkyConsumptionData>("yearly cache", REFRESH_FIRST_HOUR_OF_DAY,
+                () -> {
+                    final LocalDate today = LocalDate.now();
+                    return getConsumptionData(YEARLY, LocalDate.of(today.getYear() - 1, 1, 1), today, true);
+                });
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Linky handler.");
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.submit(this::login);
+
+        final LocalDateTime now = LocalDateTime.now();
+        final LocalDateTime nextDayFirstTimeUpdate = now.plusDays(1).withHour(REFRESH_FIRST_HOUR_OF_DAY)
+                .truncatedTo(ChronoUnit.HOURS);
+        refreshJob = scheduler.scheduleWithFixedDelay(this::updateData,
+                ChronoUnit.MINUTES.between(now, nextDayFirstTimeUpdate) % REFRESH_INTERVAL_IN_MIN + 1,
+                REFRESH_INTERVAL_IN_MIN, TimeUnit.MINUTES);
+    }
+
+    private synchronized boolean login() {
+        logger.debug("login");
+
+        LinkyConfiguration config = getConfigAs(LinkyConfiguration.class);
+        Request requestLogin = new Request.Builder().url(LOGIN_BASE_URI)
+                .post(LOGIN_BODY_BUILDER.add("IDToken1", config.username).add("IDToken2", config.password).build())
```

Comment:
Most builder designs return the same object, so I don't think you should use a builder as a static field and instead should construct it each time. Does LOGIN_BODY_BUILDER return a new builder instance each time you mutate it? If not then LOGIN_BODY_BUILDER should instead be a static method, named appropriately of course.